translation/translation.js

```diff
@@ -0,0 +1,271 @@
+                        }
+                        else {
+                            $("div#content div#edition.tab-pane div.edition div.menu-content div#edition-form-container div h3").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                            $("div#content div#edition.tab-pane div.edition div.menu-content div#edition-form-container form#jforms_view_edition.form-horizontal div.control-group label").filter(function(){
+                                return $(this).text() === italian
+                            }).html(italian);
+                        }
+                    });
+            });
+
+        }
+    }
+});
+
+//This function is executed when the lizmappopupdisplayed event is triggered
+lizMap.events.on({
+    lizmappopupdisplayed: function(e) {
+        // get browser language
+        var userLang = navigator.language || navigator.userLanguage;
+        if (userLang === 'fr' || userLang === 'fr-FR'){
+
+            var translationIT_FR3 = {};
+
+            // get the json file with transalted strings
+            // the path to the json file must be provided as relative path, using the web url the browser can return a security error
+            fetch('../../../../../../concerteaux3d/translation.json').then(function(response) {
+                if (response.ok)
+                    return response.json();
+                throw new Error('Network response was not ok.');
+                }).then(function(translation) {
+
+                    // put the json content in to the dictionary
+                    translationIT_FR3 = translation;
+                    //console.log(translationIT_FR3);
+                    // iterate over dictionary key
+                    Object.keys(translationIT_FR3).forEach((italian) => {
+                        //check if a translation is provided for each string
+                        if (translationIT_FR3[italian] != ""){
+                            // translate layers name in the popup
+                            $("div#content div#map-content div#map.olMap div#OpenLayers_Map_377_OpenLayers_ViewPort div#OpenLayers_Map_377_OpenLayers_Container div#liz_layer_popup.olPopup.lizmapPopup div#liz_layer_popup_GroupDiv div#liz_layer_popup_contentDiv.olPopupContent.lizmapPopupContent h4").filter(function(){
+                                return $(this).text() === italian
+                            }).html(translationIT_FR3[italian]);
+                            // translate aliases or fields name in the popup
+                            $("div#content div#map-content div#map.olMap div#OpenLayers_Map_377_OpenLayers_ViewPort div#OpenLayers_Map_377_OpenLayers_Container div#liz_layer_popup.olPopup.lizmapPopup div#liz_layer_popup_GroupDiv div#liz_layer_popup_contentDiv.olPopupContent.lizmapPopupContent div.lizmapPopupDiv table.lizmapPopupTable tbody tr th").filter(function(){
```

Comment:
Are you sure 377 in `#OpenLayers_Map_377_OpenLayers_Container` won't change when reloading ?

Reply:
The id `#OpenLayers_Map_377_OpenLayers_Container` is always the same in my browser, could it be browser dependent?